release-23.1: acceptance: use single-node cluster and disable metamorphic constants

rafiss opened the pull request:
Backport 1/1 commits from #112633.

/cc @cockroachdb/release

Release justification: test only change

---

When possible, the acceptance tests will now use the start-single-node command. This avoids unnecessary attempts to perform replication in many tests.

Also, disable metamorphic constants, which can slow end-to-end tests down.

fixes https://github.com/cockroachdb/cockroach/issues/112605
Release note: None

rafiss committed:
acceptance: use single-node cluster and disable metamorphic constants

When possible, the acceptance tests will now use the start-single-node
command. This avoids unnecessary attempts to perform replication in many
tests.

Also, disable metamorphic constants, which can slow end-to-end tests
down.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112800)
<!-- Reviewable:end -->

rafiss committed:
acceptance: avoid replication check for single-node clusters

The test has gotten flaky on this step, but we can avoid it for
single-node clusters.

Also use subtests to make it easier to view test results.

Release note: None

chrisseto reviewed (approved):
LGTM!
